x-pack/plugins/maps/public/connected_components/map_settings_panel/navigation_panel.tsx

```diff
@@ -4,25 +4,198 @@
+    if (settings.fixedLocation.zoom < minZoom) {
+      onFixedZoomChange(minZoom);
+    } else if (settings.fixedLocation.zoom > maxZoom) {
+      onFixedZoomChange(maxZoom);
+    }
+
+    if (settings.browserLocation.zoom < minZoom) {
+      onBrowserZoomChange(minZoom);
+    } else if (settings.browserLocation.zoom > maxZoom) {
+      onBrowserZoomChange(maxZoom);
+    }
+  };
+
+  const onInitialLocationChange = (optionId: string): void => {
+    updateMapSetting('initialLocation', optionId);
+  };
+
+  const onFixedLatChange = (event: ChangeEvent<HTMLInputElement>) => {
+    let value = parseFloat(event.target.value);
+    if (isNaN(value)) {
+      value = 0;
+    } else if (value < -90) {
+      value = -90;
+    } else if (value > 90) {
+      value = 90;
+    }
+    updateMapSetting('fixedLocation', { ...settings.fixedLocation, lat: value });
+  };
+
+  const onFixedLonChange = (event: ChangeEvent<HTMLInputElement>) => {
+    let value = parseFloat(event.target.value);
+    if (isNaN(value)) {
+      value = 0;
+    } else if (value < -180) {
+      value = -180;
+    } else if (value > 180) {
+      value = 180;
+    }
+    updateMapSetting('fixedLocation', { ...settings.fixedLocation, lon: value });
+  };
+
+  const onFixedZoomChange = (value: number) => {
+    updateMapSetting('fixedLocation', { ...settings.fixedLocation, zoom: value });
+  };
+
+  const onBrowserZoomChange = (value: number) => {
+    updateMapSetting('browserLocation', { zoom: value });
+  };
+
+  const useCurrentView = () => {
+    updateMapSetting('fixedLocation', {
+      lat: center.lat,
+      lon: center.lon,
+      zoom: Math.round(zoom),
+    });
   };
 
+  function renderInitialLocationInputs() {
+    if (settings.initialLocation === INITIAL_LOCATION.LAST_SAVED_LOCATION) {
+      return null;
+    }
+
+    const zoomFormRow = (
+      <EuiFormRow
+        label={i18n.translate('xpack.maps.mapSettingsPanel.initialZoomLabel', {
+          defaultMessage: 'Initial zoom',
+        })}
+        display="columnCompressed"
+      >
+        <ValidatedRange
+          min={settings.minZoom}
+          max={settings.maxZoom}
+          step={1}
+          value={
+            settings.initialLocation === INITIAL_LOCATION.BROWSER_LOCATION
+              ? settings.browserLocation.zoom
+              : settings.fixedLocation.zoom
+          }
+          onChange={
+            settings.initialLocation === INITIAL_LOCATION.BROWSER_LOCATION
+              ? onBrowserZoomChange
+              : onFixedZoomChange
+          }
+          showInput
+          showRange
+          compressed
+        />
+      </EuiFormRow>
+    );
+
+    if (settings.initialLocation === INITIAL_LOCATION.BROWSER_LOCATION) {
+      return zoomFormRow;
+    }
+
+    return (
+      <>
+        <EuiFormRow
+          label={i18n.translate('xpack.maps.mapSettingsPanel.initialLatLabel', {
+            defaultMessage: 'Initial latitude',
```

Comment:
this is hard to explain, but due to the auto-validation of the value, it's really hard to actually type in a latitude/longitude here


e.g. trying to type `12` in the latitude box
![type12](https://user-images.githubusercontent.com/1833023/81005803-8207d000-8e1c-11ea-8bb5-b5b455ffd180.gif)

Because the values are validated, users cannot backspace the `0`. This means a user has to find a way to type around the `0` without the validation to trigger.  If you put cursor on the left-hand side of the `0`, and you start typing, you need to be careful not to trigger the `90` validation or the value will snap back. So you need to know to put in the decimal comma _before_  the correction to 90 triggers. You need a magic touch.

Compare this to the validation-behavior in the `go to` UX.

Not sure if this is worth fixing, especially this close FF, but it feels a little off.

Reply:
discussed with @nreese will keep as is. minor issue, no need to spend time on.